Directory.Build.props

```diff
@@ -23,7 +30,7 @@
     <!-- AssemblyFileVersion should change for every build. -->
     <!-- For preview releases, sample Version = 3.0.20221104.1-preview -->
     <!-- For stable releases, sample Version = 3.0.0 -->
-    <VersionPrefix>$(Major).$(Minor).$(Build_BuildNumber)</VersionPrefix>
+    <VersionPrefix>$(Major).$(Minor).$(Patch).$(Revision)</VersionPrefix>
```

Comment:
Using the ADO build number like this was originally done for generating the ManagedBatchParser nuget package - but that number has nothing to do with the STS builds we generate so I'm changing that here. I'll be following up with a separate PR for getting the ManagedBatchParser back to using the ADO build number itself since that doesn't need to follow the STS versioning scheme.